include/host1x.h

```
@@ -161,12 +161,12 @@ int host1x_overlay_set(struct host1x_overlay *overlay,
 
 #define HOST1X_BO_CREATE(host1x, size, flags)				\
 ({									\
-	struct host1x_bo *bo = host1x_bo_create(host1x, size, flags);	\
-	if (!bo)							\
+	struct host1x_bo *_bo_ = host1x_bo_create(host1x, size, flags);	\
+	if (!_bo_)							\
 		fprintf(stderr,						\
 			"ERROR: %s:%d: host1x_bo_create() failed\n",	\
 			__FILE__, __LINE__);				\
-	bo;								\
+	_bo_;								\
 })
```

Hmpf, this is a bit nasty IMO (having to worry about outer scopes just to avoid using an inline function, underscore identifiers being reserved by the compiler)... Maybe sticking the logic inside a static-inline helper is nicer? Something like

```c
static inline struct host1x_bo *host1x_bo_create_helper(struct host1x *host1x, size_t size, int flags, const char *file, int line)
{
	struct host1x_bo *bo = host1x_bo_create(host1x, size, flags);
	if (!bo)
		fprintf(stderr,
 			"ERROR: %s:%d: host1x_bo_create() failed\n",
			file, line);
	return bo;
}
#define HOST1X_BO_CREATE(host1x, size, flags) host1x_bo_create_helper(host1x, size, flags, __FILE__, __LINE__)
```
instead?

---

I don't mind, both clunky to me :)

---

TBH, I don't really care much for those helpers myself. All they do is add some printing, they don't prevent the calling code from misbehaving as a result...

---

I don't like when something fails silently.

---

Well, maybe we should add a call to abort here in that case?

---

Won't hurt, probably.

---

However it will hurt in case of opentegra use, as it may fail easily on OOM. CMA quickly gets fragmented.

---

Yeah, but then either the caller will have to deal with this (and in that case, it should probably do the error-reporting as well), or else we'll probably crash soon anyway

---

In case of opentegra, it fallbacks to non-acceleratable allocation. I don't mind grate tests crashing, it's likely to be something much more severe than OOM.

---

Can't opentegra call the non-wrapped version of host1x_bo_create, then?

---

Just to be clear, these wrappers are not important to me, do as you please with them. If you want to keep this patch as-is, I'm fine with that. This was just a suggestion.

---

Opentegra calls grate_create_texture(), which calls the macro.

I already re-worked it based on your comments ;)

---

Right, and grate_create_texture already deals with this failing. Arguably, it wouldn't need to, it could call the un-wrapped version and print itself. But whatever ;)

---

I think the state of current error-handling in grate is good enough, at least you'll now if something gone wrong and where to start to look at. Grate is mostly a dev tool, graceful error-handling and avoiding of a tiny memleaks is always good to have but not necessary here IMO.

---

Sure, I agree.